touchforms/formplayer/api.py

```diff
@@ -274,12 +280,36 @@ def post_data(data, auth=None, content_type="application/json"):
     if auth:
         d['hq_auth'] = auth.to_dict()
     domain = d.get("domain")
-
-    if domain:
-        d['uses_sql_backend'] = use_sqlite_backend(domain)
+    if 'username' in d:
+        d['restoreAs'] = d['username']
     # just default to old server for now
-    url = settings.XFORMS_PLAYER_URL
-    return post_data_helper(d, auth, content_type, settings.XFORMS_PLAYER_URL)
+    return perform_experiment(d, auth, content_type)
+
+
+def perform_experiment(d, auth, content_type):
+    experiment = FormplayerExperiment(name=d["action"], context={'request': d})
+    with experiment.control() as c:
+        c.record(post_data_helper(d, auth, content_type, settings.XFORMS_PLAYER_URL))
+    with experiment.candidate() as c:
+        # If we should already have a session, look up its ID in the experiment mapping. it better be there.
+        # This is terrible, but we use both in different places.
+        cache = get_redis_client()
+        if "session_id" in d:
+            control_session_id = d["session_id"]
+            candidate_session_id = cache.get('touchforms-to-formplayer-session-id-%s' % control_session_id)
+            d["session_id"] = candidate_session_id
+            d["session-id"] = candidate_session_id
+            logging.info("Got candidate %s from control %s" % (candidate_session_id, control_session_id))
+        if "session-id" in d:
+            control_session_id = d["session-id"]
+            candidate_session_id = cache.get('touchforms-to-formplayer-session-id-%s' % control_session_id)
+            d["session-id"] = candidate_session_id
+            d["session_id"] = candidate_session_id
+            logging.info("Got candidate %s from control %s" % (candidate_session_id, control_session_id))
```

Comment:
there seems to be a bit of duplication in these two blocks which would be good if it was consolidated.

also, we can't guarantee that there will be a value for `candidate_session_id`. when this is released there will be open touchforms sessions with no corresponding formplayer session and when people answer the next question in those surveys it will get here with a `candidate_session_id` that is `None`